### PR TITLE
implemented support for prism monitor mode header

### PR DESCRIPTION
**Pull request by nl5887**

### CODE REVIEW

File: layers/prism.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+package layers
+
+import (
+	"encoding/binary"
+
+	"github.com/google/gopacket"
+)
+
+const ()
```
Comment (gconnell) [inline layers/prism.go]: not necessary, maybe remove?

File: layers/prism.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+func decodePrismValue(data []byte, pv *PrismValue) {
+	pv.Did = binary.LittleEndian.Uint32(data[0:4])
+	pv.Status = binary.LittleEndian.Uint16(data[4:6])
+	pv.Length = binary.LittleEndian.Uint16(data[6:8])
+	pv.Data = binary.LittleEndian.Uint32(data[8:12])
+}
+
+type PrismValue struct {
+	Did    uint32
```
Comment (gconnell) [inline layers/prism.go]: Maybe DID?

File: layers/prism.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+func (m *PrismHeader) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	m.Code = binary.LittleEndian.Uint16(data[0:4])
+	m.Length = binary.LittleEndian.Uint16(data[4:8])
+	m.DeviceName = string(data[8:24])
+	m.BaseLayer = BaseLayer{Contents: data[:m.Length], Payload: data[m.Length:len(data)]}
+
+	offset := uint16(24)
+
+	m.Values = make([]PrismValue, (m.Length-offset)/12)
```
Comment (gconnell) [inline layers/prism.go]: check that remaining data is length 12, return error if not?

File: layers/prism_test.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
+// Copyright 2014, Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	_ "fmt"
```
Comment (gconnell) [inline layers/prism_test.go]: I think this can be removed?

File: layers/prism.go (hunk 2 above)
Comment (gconnell) [inline layers/prism.go]: Also, DID should probably be an enum.  Maybe something like:

type PrismDID uint32
const (
  PrismHostTime PrismDID = X
  PrismMACTime PrismDID = Y
  ...
)

**Comment by gconnell** [comment]
Thanks for the commit!  Just a few minor comments, and I think we'll be good to go.

**Comment by nl5887** [comment]
Our Wireshark friends helped me out again, apparently there are two types (type1 and type2). I was looking at an old (or different spec) Implemented both the type1 and type2 Prism DID types.

**Pull request merged**